skt/__init__.py

```diff
@@ -476,16 +477,38 @@ def __init__(self, path, basecfg, cfgtype=None, makeopts=None,
         self.basecfg = os.path.expanduser(basecfg)
         self.cfgtype = cfgtype if cfgtype is not None else "olddefconfig"
         self._ready = 0
-        self.makeopts = None
         self.buildlog = "%s/build.log" % self.path
         self.defmakeargs = ["make", "-C", self.path]
         self.enable_debuginfo = enable_debuginfo
 
         if makeopts is not None:
-            # FIXME: Might want something a bit smarter here, something that
-            # would parse it the same way bash does
-            self.makeopts = makeopts.split(' ')
-            self.defmakeargs += self.makeopts
+            self.makeopts = self.defmakeargs + shlex.split(makeopts)
+        else:
+            self.makeopts = self.defmakeargs
+
+        # If the user provided a directory to hold kernel build artifacts,
+        # we must add the appropriate make options.
+        if kbuild_output is not None:
+            # Ensure the path is absolute, even if the user provided a path
+            # relative to their home directory with a tilde (~).
+            self.kbuild_output_dir = os.path.abspath(
+                os.path.expanduser(kbuild_output)
+            )
+            logging.info(
+                "Found kernel build output directory specifier in make "
+                "options: {}".format(self.kbuild_output_dir)
+            )
+
+            # Ensure the output directory exists
+            # NOTE(mhayden): os.makedirs will cause an error if the directory
+            # already exists, so we must check for the directory first.
+            if not os.path.isdir(self.kbuild_output_dir):
+                os.makedirs(self.kbuild_output_dir)
```

Comment:
Is this a problem?  If the directory already exists, does that mean another container is using it?  Is it clean?  An existing directory on shared storage makes me wonder.

Reply:
It catches a situation where the directory might exist because someone made it manually or mounted a filesystem on that directory.

Reply:
That was my point.  Is that going to lead to collision with other work?

Reply:
This is an implementation issue. ;) One would hope that the user would ensure that output directories are not shared.

Within OpenShift, the kernel source is kept inside an ephemeral volume within the container and the build output goes into storage with a guaranteed size allotment that persists after the container is gone.